### ISSUE TITLE
Error CS0656 is incomprehensible and misleading.

### ISSUE THREAD

**Issue by CornedBee** [dotnet/roslyn#25465]
**Version Used**: .Net Core 2.1.100

When you compile code that uses dynamic, the compiler generates calls to various support functions. These functions are defined in the Microsoft.CSharp package. If this package isn't referenced, it generates CS0656.

This error, its message, and its help page are remarkably unhelpful.

> error CS0656: Missing compiler required member 'Microsoft.CSharp.RuntimeBinder.Binder.SetMember'

OK, so it is missing some member. As a new programmer, I'm baffled. Am I supposed to implement this? Is my installation corrupted? Why doesn't the language just work?

But VS allows me to get more information, by putting the cursor on the error in the output window and pressing F1. This leads me here:

https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0656?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev15.query%3FappId%3DDev15IDEF1%26l%3DEN-US%26k%3Dk(CS0656)%3Bk(vs.output)%3Bk(SolutionItemsProject)%26rd%3Dtrue

This page suggests that one of two things is happening, and I quote:

- Your installation of the common language runtime is corrupt.
- You have a reference to an assembly that defines a type that is also found in the common language runtime. However, your assembly's type is not defined the way the C# compiler expects.

I check my references. Nope, not doing anything like the second option says. In fact, my test project references nothing except NETStandard. Nor does the project itself contain anything like it.

So it's gotta be the first option, right? My installation is corrupt? Should I reinstall? Will it work then? (Hint: it won't.)

So thankfully I'm actually experienced enough to not reinstall, and first search the web a little more. Which leads me to the actual problem (not referencing Microsoft.CSharp) and the solution (installing it). But the user experience in this case was abysmal. Furthermore, this error appeared only after my latest round of dependency upgrades (I suspect I was previously indirectly referencing the package), so the code worked previously "out of the box".

---------------

The compiler error should be more helpful than that. Modern compilers usually give solution hints (see: Clang, rustc, many more). In this case, "A language feature package may be missing." would be a good hint even if you don't want to map from the missing function to the probably containing package.

The help page for the error is just inacceptable. I have added a comment with the real fix, but this should be in a most prominent place of the actual text.

**Comment by svick** [comment]
As far as I can see, there are two problems here:

1. The compiler error is hard to understand.
2. The documentation page is incorrect.

This repo should be the right one for the first problem, but I think you should open a separate issue in [the dotnet/docs repo](https://github.com/dotnet/docs) for the second problem, because that's [where the source for that page is](https://github.com/dotnet/docs/blob/master/docs/csharp/misc/cs0656.md).
